pkg/common/patcher/patcher.go

```diff
@@ -25,6 +25,8 @@ type PatchClient[R runtime.Object] interface {
 type Patcher[R runtime.Object, Sp any, St any] interface {
 	AddFinalizer(context.Context, R, ...string) (bool, error)
 	RemoveFinalizer(context.Context, R, ...string) error
```

Comment:
actually, could we add opts var in the RemoveFinalizer func

Reply:
you mean the patch option for the `p.client.Patch(ctx, accessor.GetName(), types.MergePatchType, patchBytes, metav1.PatchOptions{})`？ 

we will not set `force` to `true` for the patch option, it is used for `re-acquire conflicting fields owned by other people.`
in our case, we just need to keep the `resourceVersion` empty.

Reply:
not necessary to be a mevat.PatchOptions, but a options field that you can know whether it is force patch or not.

Reply:
added a PatchOptions param for the RemoveFinalizers func.